### PR TITLE
Perform an overall code cleanup

### PR DESCRIPTION
**Pull request by lppedd**
As per title, this is just a cleanup to enhance readability or remove unused/duplicated APIs.

I've split each cleanup action into its own commit to make it easier to review them.

### CODE REVIEW

**Comment by lppedd** [comment]
CI seems to be failing because of unreachable codecov.

File: src/commonMain/kotlin/io/github/petertrr/diffutils/DiffUtils.kt [hunk 1]
```diff
@@ -16,187 +16,154 @@
...
+ * By default, uses the Myers algorithm.
+ *
+ * @param sourceText The original text
+ * @param targetText The target text
+ * @param algorithm The diff algorithm to use
+ * @param progress The diff algorithm progress listener
+ * @param includeEqualParts Whether to include equal data parts into the patch. `false` by default.
+ * @return The patch describing the difference between the original and target text
  */
+@JvmOverloads
```
Comment (petertrr) [inline src/commonMain/kotlin/io/github/petertrr/diffutils/DiffUtils.kt:48]: :+1:

**Review by petertrr — COMMENTED** [review]
Looks nice! Even though I could argue that keeping one class per file is really Java-style, I don't mind extracting some things.

Please take a look at my comments.

BTW, I update codecov action on main, hopefully it will be more stable if you rebase.

File: src/commonMain/kotlin/io/github/petertrr/diffutils/patch/Patch.kt [hunk 2]
```diff
@@ -23,111 +23,125 @@ import io.github.petertrr.diffutils.algorithm.Change
...
         val result = target.toMutableList()
         val it = deltas.listIterator(deltas.size)
+
         while (it.hasPrevious()) {
             val delta = it.previous()
-            delta.verifyAndApplyTo(result).takeIf { it != VerifyChunk.OK }?.let {
-                conflictOutput.processConflict(it, delta, result)
-            }
+            val verifyChunk = delta.verifyAndApplyTo(result)
+            conflictOutput.processConflict(verifyChunk, delta, result)
```
[the remark this answers is not in the document]
Comment (lppedd) [reply src/commonMain/kotlin/io/github/petertrr/diffutils/patch/Patch.kt:50]: Oh this has been done on purpose. I've delegated the check to the implementors of `ConflictOutput`. So if you look inside `processConflict` you'll see I check for `VerifyChunk.OK`.

The rationale is `ConflictOutput` shouldn't know we've filtered for `VerifyChunk.OK` beforhand, and take it for granted.

File: src/commonTest/kotlin/io/github/petertrr/diffutils/DiffUtilsTest.kt [hunk 3]
```diff
@@ -18,16 +18,10 @@
  */
 package io.github.petertrr.diffutils
 
-import io.github.petertrr.diffutils.patch.ChangeDelta
-import io.github.petertrr.diffutils.patch.Chunk
-import io.github.petertrr.diffutils.patch.DeleteDelta
-import io.github.petertrr.diffutils.patch.EqualDelta
-import io.github.petertrr.diffutils.patch.InsertDelta
-import io.github.petertrr.diffutils.patch.Patch
+import io.github.petertrr.diffutils.patch.*
```
[the remark this answers is not in the document]
Comment (lppedd) [reply src/commonTest/kotlin/io/github/petertrr/diffutils/DiffUtilsTest.kt:21]: I think it has a threshold after which it doesn't complain anymore. I'll restore it tho.

File: src/commonMain/kotlin/io/github/petertrr/diffutils/algorithm/myers/MyersDiff.kt [hunk 4]
```diff
@@ -26,106 +26,126 @@ import io.github.petertrr.diffutils.patch.DeltaType
...
+    private fun buildRevision(actualPath: PathNode): List<Change> {
+        var path = if (actualPath.snake) {
+            actualPath.prev
+        } else {
+            actualPath
         }
+
+        val changes = ArrayList<Change>()
+
+        // This can be improved to avoid the non-null assertion on prev
```
[the remark this answers is not in the document]
Comment (lppedd) [reply src/commonMain/kotlin/io/github/petertrr/diffutils/algorithm/myers/MyersDiff.kt:125]: Yup! I did initially use a TODO, but Detekt doesn't like them apparently.

File: src/commonMain/kotlin/io/github/petertrr/diffutils/algorithm/myers/PathNode.kt [hunk 5]
```diff
@@ -20,54 +20,63 @@ package io.github.petertrr.diffutils.algorithm.myers
...
         return if (!snake && prev != null) {
             prev.previousSnake()
         } else {
             this
         }
     }
 
-    override fun toString() = generateSequence(this) { it.prev }
-        .joinToString(prefix = "[", postfix = "]") { "(${it.i}, ${it.j})" }
+    override fun toString(): String {
```
[the remark this answers is not in the document]
Comment (lppedd) [reply src/commonMain/kotlin/io/github/petertrr/diffutils/algorithm/myers/PathNode.kt:66]: Mainly two reasons:
1. like with one class per file, an implementation closer to the upstream Java one helps in porting future changes
2. we need to keep in mind readability. While `generateSequence + joinToString` is definitely more kotlinesque, understanding what's happening when you approach it is much more challenging. With a sequential solution you immediately infer what the result will be.

File: src/commonMain/kotlin/io/github/petertrr/diffutils/algorithm/myers/MyersDiff.kt [hunk 6]
```diff
@@ -26,106 +26,126 @@ import io.github.petertrr.diffutils.patch.DeltaType
...
-     * Return empty diff if get the error while procession the difference.
+     * Returns an empty diff if we get an error while procession the difference.
      */
-    override fun computeDiff(source: List<T>, target: List<T>, progress: DiffAlgorithmListener?): List<Change> {
-        progress?.diffStart()
-        val path = buildPath(source, target, progress)
+    override fun computeDiff(source: List<T>, target: List<T>, progress: DiffAlgorithmListener): List<Change> {
+        progress.diffStart()
+
+        val path = buildPath(source, target, progress) ?: error("Expected a non-null path node")
```
[the remark this answers is not in the document]
Comment (lppedd) [reply src/commonMain/kotlin/io/github/petertrr/diffutils/algorithm/myers/MyersDiff.kt:36]: I think this is an unexpected state, which means a programming error, and not a diff problem.

File: src/commonMain/kotlin/io/github/petertrr/diffutils/algorithm/myers/MyersDiff.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (petertrr) [reply src/commonMain/kotlin/io/github/petertrr/diffutils/algorithm/myers/MyersDiff.kt:125]: Hm, I see. I'm okay to disable the `ForbiddenComment` rule for now. Probably, in the future I can add a baseline file to track known code smells

File: src/commonMain/kotlin/io/github/petertrr/diffutils/patch/Patch.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (petertrr) [reply src/commonMain/kotlin/io/github/petertrr/diffutils/patch/Patch.kt:50]: Ah, OK, didn't notice it.

File: src/commonMain/kotlin/io/github/petertrr/diffutils/text/DiffRow.kt [hunk 7]
```diff
@@ -28,4 +28,7 @@ public data class DiffRow(val tag: Tag, val oldLine: String, val newLine: String
         CHANGE,
         EQUAL,
     }
+
+    override fun toString(): String =
+        "[$tag, $oldLine, $newLine]"
```
Comment (lppedd) [inline src/commonMain/kotlin/io/github/petertrr/diffutils/text/DiffRow.kt:33]: Context: done in the same way as the Java one.

**Review by petertrr — APPROVED** [review]
LGTM!

**Comment by lppedd** [comment]
Thank you! I'll try to get code coverage up with one of the following PRs to port the java-diff changes.

**Pull request merged**